sys/mips/include/cherireg.h

```diff
@@ -95,29 +90,10 @@
 #define	CHERI128_PERM_SW2			(1 << 17)	/* 0x00020000 */
 #define	CHERI128_PERM_SW3			(1 << 18)	/* 0x00040000 */
 
-#if (CHERICAP_SIZE == 32)
-#define	CHERI_PERM_SW0		CHERI256_PERM_SW0
-#define	CHERI_PERM_SW1		CHERI256_PERM_SW1
-#define	CHERI_PERM_SW2		CHERI256_PERM_SW2
-#define	CHERI_PERM_SW3		CHERI256_PERM_SW3
-#define	CHERI_PERM_SW4		CHERI256_PERM_SW4
-#define	CHERI_PERM_SW5		CHERI256_PERM_SW5
-#define	CHERI_PERM_SW6		CHERI256_PERM_SW6
-#define	CHERI_PERM_SW7		CHERI256_PERM_SW7
-#define	CHERI_PERM_SW8		CHERI256_PERM_SW8
-#define	CHERI_PERM_SW9		CHERI256_PERM_SW9
-#define	CHERI_PERM_SW10		CHERI256_PERM_SW10
-#define	CHERI_PERM_SW11		CHERI256_PERM_SW11
-#define	CHERI_PERM_SW12		CHERI256_PERM_SW12
-#define	CHERI_PERM_SW13		CHERI256_PERM_SW13
-#define	CHERI_PERM_SW14		CHERI256_PERM_SW14
-#define	CHERI_PERM_SW15		CHERI256_PERM_SW15
-#else /* (!(CHERICAP_SIZE == 32)) */
 #define	CHERI_PERM_SW0		CHERI128_PERM_SW0
 #define	CHERI_PERM_SW1		CHERI128_PERM_SW1
 #define	CHERI_PERM_SW2		CHERI128_PERM_SW2
 #define	CHERI_PERM_SW3		CHERI128_PERM_SW3
```

Comment:
Why not nuke the 256 definitions above and make the CHERI128 ones CHERI directly, rather than the extra indirection?

Reply:
Oh, yes, and I also didn't remove the CHERI256_PERM constants.  I will simplify this to match what I did for RISC-V.